### DISCUSSION TITLE
How exactly does this work with *arr?

### DISCUSSION THREAD

**Discussion by varzaman** [haridusministeerium/docker-seedbox-rclone-fetch-extract#2] [General]
Hello,

Was hoping something could be cleared up for me.

So I understand the idea of this docker image. What I don't fully understand is how it 100% works with the *arr apps, maybe I have a wrong assumption.

Here's my understanding on how *arr processes downloads.

1. A client that sonarr is monitoring marks a torrent as complete. These files exist in a specific directory.
2. Sonarr sees that the torrent client marked a torrent as complete. It then looks into the directory
3. It sees the files are there, and starts importing them.
 

With this image, wouldn't the remote torrent file be marked as complete, but then nothing actually exists in the local folder (as nothing has been downloaded yet)? When Sonarr goes to look at the local directory, it won't see anything, and thus not import.

I feel like I'm missing something here.

**Comment by laur89** [comment]
Your assumptions are all correct, but the \*arr import doesn't end there. After having checked the expected asset directory and found it empty, it will retry later. Meanwhile the service downloads the asset from server (initially to a directory \*arrs don't see so it can't do a partial import) and then moves the files to expected directory. And eventually it gets imported.

**Reply by varzaman** [reply]
"After having checked the expected asset directory and found it empty, it will retry later."

This is the part of the puzzle I was missing, thanks for responding. 

I'm planning out my server while waiting for all my parts to arrive so I wasn't able to check it myself, so really appreciate it.
